source/slang/slang-ir-autodiff-cfg-norm.cpp

```diff
@@ -133,13 +160,28 @@ struct CFGNormalizationPass
         return false;
     }
 
+    void _moveVarsToRegionHeader(BreakableRegionInfo* region, IRBlock* block)
+    {
+        for (auto child = block->getFirstChild(); child;)
+        {
+            auto nextChild = child->getNextInst();
+
+            if (as<IRVar>(child))
+            {
+                child->insertBefore(region->headerBlock->getTerminator());
+            }
+
+            child = nextChild;
+        }
+    }
 
     RegionEndpoint getNormalizedRegionEndpoint(
         BreakableRegionInfo* parentRegion,
         IRBlock* entryBlock,
         List<IRBlock*> afterBlocks)
     {
         IRBlock* currentBlock = entryBlock;
+        _moveVarsToRegionHeader(parentRegion, currentBlock);
```

Comment:
We should just fix the `eliminatePhi` logic so we don't have to handle this here?